dag_in_context/src/optimizations/loop_unroll.egg

```diff
@@ -23,3 +23,44 @@
           outputs) ;; right now, loop unrolling shares the same outputs, but we could add more context here
         (Arg inputs-ty (NoContext)))))
  :ruleset loop-peel)
+
+
+(function UnrollCtx (Expr Expr) Assumption)
+
+;; unroll a loop with constant bounds and initial value
+(rule
```

Comment:
This rule looks reasonable. Question though:

I was imagining that loop unrolling would do something like:

```
do {
   <body>
} while (<pred>)
=>
do {
  <body>
  if (<pred>) <body>
} while(<pred>)
```

I think we'd be able to rewrite bounded-iteration loops this way, and we'd match more often. If we really are comparing (i < C) and (i + 1 < C) ... we'd be able to simplify predicates as well I'd imagine.

Why are we doing this one that hard-codes a multiple of 4? I suspect there's a subtlety I'm missing.

Reply:
I'd like to make it work that way instead- I decided to do the simplest thing first